### ISSUE TITLE
Convert README and other docs to markdown

### ISSUE THREAD

**Issue by lukebakken** [pika/pika#1073]
[PyPi finally supports markdown](https://dustingram.com/articles/2018/03/16/markdown-descriptions-on-pypi) so we should consider moving from `.rst` since markdown is more familiar (myself included).

**Comment by giovana-morais** [comment]
can I help with this?

**Comment by lukebakken** [comment]
Of course, we review all PRs as time allows. Thank you for offering! Please base your work on the `stable` branch and we can forward-port it to `master`

**Comment by lukebakken** [comment]
It's no problem, we're not in a hurry. I will check out `toctrees` ... I've never heard of them.

**Comment by romantolkachyov** [comment]
It is wrong to use markdown with Sphinx. Don't touch `docs/` :-) 

It is harder to maintain markdown with rst inclusions (in this case you must know both syntaxes and know how to use them together) rather than native rst.

**Comment by lukebakken** [comment]
@romantolkachyov the plan is to only use markdown. rst isn't as familiar to this project's maintainers (myself included) and there are zero volunteers to help when rst issues come up.

**Comment by romantolkachyov** [comment]
@lukebakken I can help you with rst. Just mention me.

You will have more issues when you try use rst syntax (directives) in markdown if you want extract docs from sources or make other advanced things. sphinx/rst is de facto standard in python, I think we must follow it.

Just for example (quick googling): http://ericholscher.com/blog/2016/mar/15/dont-use-markdown-for-technical-docs/ And there was a lot more pain when I've tried it last time (2-3 years ago) but I didn't remember exactly reason. 

It just a recommendation anyway.
